src/TiledArray/math/linalg/rank-local.cpp

```diff
@@ -23,26 +23,33 @@
  */
 
 #include <TiledArray/math/linalg/rank-local.h>
+#include <TiledArray/math/lapack.h>
+
+template<class F, typename ... Args>
+inline int ta_lapack_call(F f, Args ... args) {
+  lapack_int info;
+  auto ptr = [](auto &&a) {
+    using T = std::remove_reference_t<decltype(a)>;
+    if constexpr (std::is_pointer_v<T>) return a;
+    else return &a;
+  };
+  f(ptr(args)..., &info);
+  return info;
+}
 
-#include <lapacke.h>
-
-#define TA_LAPACKE_THROW(F) throw std::runtime_error("lapacke::" #F " failed")
+#define TA_LAPACK_ERROR(F) throw std::runtime_error("lapack::" #F " failed")
 
-#define TA_LAPACKE_CALL(F, ARGS...)         \
-  (LAPACKE_##F(LAPACK_COL_MAJOR, ARGS) == 0 || (TA_LAPACKE_THROW(F), false))
+#define TA_LAPACK_CALL(F, ARGS...) \
+  ((ta_lapack_call(F##_, ARGS) == 0) || (TA_LAPACK_ERROR(F), 0))
 
-/// TA_LAPACKE(fn,args) can be called only from template context, with `T`
+/// TA_LAPACK(fn,args) can be called only from template context, with `T`
 /// defining the element type
-#define TA_LAPACKE(name, args...) {                                            \
+#define TA_LAPACK(name, args...) {                                             \
   using numeric_type = T;                                                      \
   if constexpr (std::is_same_v<numeric_type, double>)                          \
-    TA_LAPACKE_CALL(d##name, args);                                            \
+                 TA_LAPACK_CALL(d##name, args);                                \
   else if constexpr (std::is_same_v<numeric_type, float>)                      \
-    TA_LAPACKE_CALL(s##name, args);                                            \
-  else if constexpr (std::is_same_v<numeric_type, std::complex<double>>)       \
-    TA_LAPACKE_CALL(z##name, args);                                            \
-  else if constexpr (std::is_same_v<numeric_type, std::complex<float>>)        \
-    TA_LAPACKE_CALL(c##name, args);                                            \
+                 TA_LAPACK_CALL(s##name, args);                                \
```

Comment:
why no complex?

Reply:
compile issues due to constexpr defificencies.